pkg/deploy/api/validation/validation.go

```diff
@@ -39,6 +39,9 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		originalContainerImageNames := getContainerImageNames(spec.Template)
+		defer setContainerImageNames(spec.Template, originalContainerImageNames)
```

Comment:
Is defer working different or will the deferred function move in the stack once  `defer` in encountered in the code? If the latter, I don't see any safety problem here, we just save a defer call that is redundant.
```go
originalContainerImageNames := getContainerImageNames(spec.Template)
handleEmptyImageReferences(spec.Template, spec.Triggers)
defer setContainerImageNames(spec.Template, originalContainerImageNames)
```
vs
```go
originalContainerImageNames := getContainerImageNames(spec.Template)
updated := handleEmptyImageReferences(spec.Template, spec.Triggers)
if updated {
   defer setContainerImageNames(spec.Template, originalContainerImageNames)
}
```